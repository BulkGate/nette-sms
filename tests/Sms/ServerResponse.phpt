<?php

/**
 * Test: Nette\Sms\ServerResponse
 */

namespace Test;

use Nette;
use Nette\Sms\Compress;
use Nette\Sms\ServerResponse;
use Tester\Assert;

require __DIR__ . '/../bootstrap.php';
require __DIR__ . '/../../src/Sms/ServerResponse.php';
require __DIR__ . '/../../src/Sms/Compress.php';

$response = new ServerResponse([
	'status' => '-1',
	'type' => 'error',
	'message' => 'Unknown action',
	'compress' => false,
	'data' => [],
]);

Assert::same(-1, $response->getStatus());

Assert::same('error', $response->getType());

Assert::same('error', $response->getType());

Assert::same('Unknown action', $response->getMessage());

Assert::same([], $response->getData());

$response = new ServerResponse([
	'status' => '-1',
	'type' => 'error',
	'message' => 'Unknown action',
	'compress' => true,
	'data' => Compress::compress(['123' => 'abc']),
]);

Assert::same(['123' => 'abc'], $response->getData());
