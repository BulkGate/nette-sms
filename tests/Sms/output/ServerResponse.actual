(object) /* #307f */ array(
	'123' => 'abc',
)
