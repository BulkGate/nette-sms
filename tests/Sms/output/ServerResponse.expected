array(123 => 'abc')
